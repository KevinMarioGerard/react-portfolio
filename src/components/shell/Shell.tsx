import * as React from 'react';
import Nav from '../nav/Nav';
import Main from './Main';

type Statics = {
    Nav: typeof Nav;
    Main: typeof Main;
};

const Shell: React.FC & Statics = props => <div className="shell">{props.children}</div>;

Shell.Nav = Nav;
Shell.Main = Main;

export default Shell;
