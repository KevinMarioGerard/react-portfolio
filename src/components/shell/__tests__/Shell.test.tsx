import { render } from '@testing-library/react';
import Shell from '../Shell';
import * as React from 'react';

describe('Shell Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(
                <Shell>
                    <Shell.Nav />
                    <Shell.Main />
                </Shell>
            );
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(
            <Shell>
                <Shell.Nav />
                <Shell.Main />
            </Shell>
        );

        expect(container).toMatchSnapshot();
    });
});
