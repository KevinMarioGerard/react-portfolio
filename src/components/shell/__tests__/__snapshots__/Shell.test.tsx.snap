// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shell Component tests should match snapshot 1`] = `
<div>
  <div
    class="shell"
  >
    <nav
      class="shell__nav"
    />
    <main
      class="shell__main"
    >
      <div
        class="shell__main__container"
      />
    </main>
  </div>
</div>
`;
